### PR TITLE
feat: add notification controls

### PR DESCRIPTION
**Pull request by KrzysztofMoch**
## Summary
Add `showNotificationControls` prop to Android & iOS

### Motivation
Allow to control video from notification controls

## Test plan
- [x] Tested locally
- [x] CI passes

### CODE REVIEW

**Comment by zoriya** [comment]
Speaking for android: why do you use custom Notifications with custom actions instead of using the standard `MediaSessionCompat` and `MediaSessionConnector`? Those two classes create system's media notifications and listen to hardware volume/media buttons (like headphones or remotes) and assistant on TV.

**Comment by KrzysztofMoch** [comment]
I want in feature to allow set custom actions

**Comment by zoriya** [comment]
You can use custom actions with the built-in media session classes: https://developer.android.com/codelabs/supporting-mediasession#4

I believe the built-in one has a huge value, especially to run react-native-video on android tv.

File: android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            notificationManager.createNotificationChannel(
+                NotificationChannel(
+                    NOTIFICATION_CHANEL_ID,
+                    NOTIFICATION_CHANEL_ID,
+                    NotificationManager.IMPORTANCE_LOW
+                )
+            )
+        }
+
+        val contentTitle = session.player.currentMediaItem?.mediaMetadata?.title ?: "Video with no title"
```
Comment (zoriya) [inline android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt]: I think we should be able to customize this via a prop and not take the one from the video's metadata.

**Comment by KrzysztofMoch** [comment]
👀 thanks for info, will check out this

File: android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt (hunk 1 above)
Comment (KrzysztofMoch) [reply android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt]: Something like prop "metadata" in source ?

Comment (zoriya) [reply android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt]: I forgot about these props, sorry. I think all of them could be used then!

Comment (freeboub) [reply android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt]: I agree, it should be useful to add a metadata prop. 
in source you already have a metadata field: https://react-native-video.github.io/react-native-video/component/props#overriding-the-metadata-of-a-source
But I am not sure the source is the good place for metadata. Metadata can change while watching live content
BTW, hardcoded string should be forbidden ! what about none english speakers :)

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 2]
```diff
@@ -347,6 +358,12 @@ public void onHostDestroy() {
         cleanUpResources();
     }
 
+    @Override
+    protected void onDetachedFromWindow() {
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:364]: why did you add this ? Is this really mandatory ?

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 3]
```diff
@@ -734,6 +755,55 @@ private void finishPlayerInitialization() {
...
+        if (showNotificationControls & player != null) {
+            playbackServiceConnection = new ServiceConnection() {
+                @Override
+                public void onServiceConnected(ComponentName name, IBinder service) {
+                    playbackServiceBinder = (PlaybackServiceBinder) service;
+
+                    try {
+                        playbackServiceBinder.getService().registerPlayer(player);
+                    } catch (Exception e) {
+                        DebugLog.e("ExoPlayer Playback", "Cloud not register ExoPlayer");
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java]: can you use a TAG in DebugLog like other DebugLog calls please ?

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 4]
```diff
@@ -734,6 +755,55 @@ private void finishPlayerInitialization() {
...
+                    }
+                }
+
+                @Override
+                public void onServiceDisconnected(ComponentName name) {
+                    playbackServiceBinder = null;
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java]: can you playbackServiceBinder.getService().unregisterPlayer(player); here ?
I am afraid the service keep holding a player Reference...
Not sure my comment make sens ...

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 5]
```diff
@@ -734,6 +755,55 @@ private void finishPlayerInitialization() {
...
+                }
+
+                @Override
+                public void onNullBinding(ComponentName name) {
+                    DebugLog.e("ExoPlayer Playback", "Cloud not register ExoPlayer");
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java]: same comment on log

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java (hunk 2 above)
Comment (KrzysztofMoch) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:364]: I don't think so, but is there better way to call some code when view id unmounted ?

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 6]
```diff
@@ -734,6 +755,55 @@ private void finishPlayerInitialization() {
...
+            if (Build.VERSION.SDK_INT >= 29) {
+                flags = Context.BIND_AUTO_CREATE | Context.BIND_INCLUDE_CAPABILITIES;
+            } else {
+                flags = Context.BIND_AUTO_CREATE;
+            }
+
+            themedReactContext.bindService(intent, playbackServiceConnection, flags);
+        }
+
+        if(!showNotificationControls && playbackServiceBinder != null) {
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java]: That is strange to setupPlaybackService in a setup function ...
maybe you should split the function in 2 different functions ?

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 7]
```diff
@@ -1934,6 +2012,20 @@ public void setContentStartTime(int contentStartTime) {
...
+    public void setShowNotificationControls(boolean showNotificationControls) {
+        this.showNotificationControls = showNotificationControls;
+
+        if (playbackServiceConnection == null && showNotificationControls) {
+            setupPlaybackService();
+        }
+
+        if(!showNotificationControls && playbackServiceBinder != null) {
+            if (player != null) {
+                playbackServiceBinder.getService().unregisterPlayer(player);
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java]: I think you should also clean playbackServiceBinder = null here ?
see my comment in setupPlaybackService(); split

File: android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+    // Player Registry
+
+    fun registerPlayer(player: ExoPlayer) {
+        if (mediaSessionsList.containsKey(player)) {
+            return
+        }
+
+        val mediaSession = MediaSession.Builder(this, player)
+            .setId("RNVideoPlaybackService_" + player.hashCode())
+            .setCallback(VideoPlaybackCallback(10000L))
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt]: maybe extract this 10 sec to a constant and add a comment indicating this is hardcoded. 10sec is OK for me

File: android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    @SuppressLint("PrivateResource")
+    private val seekForwardBtn = CommandButton.Builder()
+        .setDisplayName("forward")
+        .setSessionCommand(commandSeekForward)
+        .setIconResId(androidx.media3.ui.R.drawable.exo_notification_fastforward)
+        .build()
+
+    @SuppressLint("PrivateResource")
+    private val seekBackwardBtn = CommandButton.Builder()
+        .setDisplayName("forward")
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt]:
```suggestion
        .setDisplayName("backward")
```

File: examples/basic/src/VideoPlayer.tsx [hunk 10]
```diff
@@ -819,6 +819,7 @@ class VideoPlayer extends Component {
     return (
       <TouchableOpacity style={viewStyle}>
         <Video
+          showNotificationControls={true}
```
Comment (freeboub) [inline examples/basic/src/VideoPlayer.tsx]: can you add a button to enable / disable it please ?

File: ios/Video/NowPlayingInfoCenterManager.swift [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+        let commandCenter = MPRemoteCommandCenter.shared()
+
+        invalideTargets(commandCenter)
+
+        playTarget = commandCenter.playCommand.addTarget { [weak self] _ in
+            guard let self, let player = self.currentPlayer else {
+                return .commandFailed
+            }
+
+            if player.rate == 0 {
```
Comment (freeboub) [inline ios/Video/NowPlayingInfoCenterManager.swift:130]: That is sad to test rate to play ...
I think there is a "global issue" mixing rate and pause/playback state ...

File: ios/Video/NowPlayingInfoCenterManager.swift [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+            }
+
+            return .success
+        }
+
+        skipBackwardTarget = commandCenter.skipBackwardCommand.addTarget { [weak self] _ in
+            guard let self, let player = self.currentPlayer else {
+                return .commandFailed
+            }
+            let newTime = player.currentTime() - CMTime(seconds: 10, preferredTimescale: .max)
```
Comment (freeboub) [inline ios/Video/NowPlayingInfoCenterManager.swift]: same comment than for android and the 10 sec

File: ios/Video/RCTVideo.swift [hunk 13]
```diff
@@ -1057,6 +1079,17 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
         }
     }
 
+    @objc
+    func setShowNotificationControls(_ showNotificationControls: Bool) {
+        if showNotificationControls == true, let player = _player {
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift]: I am sure you can do cleaner code by refactoring the 2 `let player = _player` !

**Review by freeboub — CHANGES_REQUESTED** [review]
A lot of small comments, but nothing big to change.
I am just doubting about metadata source. I have never see on my project video with metadata. I really think we should fallback on field provided by the application.
You have 2 choices: 
- either use metadata from the source (quick low impact, but not dynamic during playback, and not yet implemented on android)
- either create a new metadata field in the props to allow supporting dynamic update. 

And thank you for the PR !

**Comment by KrzysztofMoch** [comment]
@zoriya seems like `MediaSessionConnector` from exoplayer is deprecated in favour of `MediaSession` from media3
CleanShot 2024-05-01 at 12 35 44

I will check if with current implementation, tv pilots will work (I will test it via adb)

**Comment by zoriya** [comment]
Yes, I tried to use MediaSession from media3 in the draft PR. Is there a reason we can't use it?

**Comment by KrzysztofMoch** [comment]
We don't have exoplayer2 any more as we migrated to media3

**Comment by KrzysztofMoch** [comment]
I have tested and with current implementation we support key events like 126, 85 so I guess with this TV should be supported ?

**Comment by zoriya** [comment]
Yes, since everything was migrated to media3 I don't see why we can't simply use: `androidx.media3:media3-session:$media3_version`

**Comment by KrzysztofMoch** [comment]
We are using it
https://github.com/react-native-video/react-native-video/blob/edbb3b60aab0085ccd047f69bd04e7898a0b62a1/android/build.gradle#L226-L229

And I am using `MediaSession` from `androidx.media3:media3-session` (you can look into imports in `VideoPlaybackService`) as `MediaSessionConnector` is no longer available in media3

**Comment by zoriya** [comment]
Okay, I'm sorry I was referring to outdated docs. I got the right one now.

I don't understand why you create a notification manually. media3's session support notifications out of the box: https://developer.android.com/media/media3/session/background-playback#notification

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java (hunk 2 above)
Comment (freeboub) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:364]: OK, no problem

**Comment by KrzysztofMoch** [comment]
@freeboub I don't see a way to dynamically update metadata... for now I am using `player.replaceMediaItem` but this resets video. I also have tried to update notification content when setting metadata but "content" is lost when you do any action (like pause, seek)

**Comment by freeboub** [comment]
> @freeboub I don't see a way to dynamically update metadata... for now I am using `player.replaceMediaItem` but this resets video. I also have tried to update notification content when setting metadata but "content" is lost when you do any action (like pause, seek)

Ok, thank you for double checking !

**Comment by KrzysztofMoch** [comment]
So I guess in this case will be better to move prop `metadata` to source ? or should I leave it outside

**Comment by freeboub** [comment]
> So I guess in this case will be better to move prop `metadata` to source ? or should I leave it outside

🤔 It should be in source then... As you will need it at playback startup it fully make sense to put it in the source !

File: ios/Video/NowPlayingInfoCenterManager.swift (hunk 11 above)
Comment (KrzysztofMoch) [reply ios/Video/NowPlayingInfoCenterManager.swift:130]: According to Apple documentation 
CleanShot 2024-05-06 at 10 59 27
So I think that checking `rate == 0` to check if player is paused in good practice

File: android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt (hunk 1 above)
Comment (KrzysztofMoch) [reply android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt]: Sure, I removed this - this now should be set by `metadata` in `source` prop

**Comment by KrzysztofMoch** [comment]
> I don't understand why you create a notification manually. media3's session support notifications out of the box

When I was using those out of the box, there could be only 1 "notification" even with there was more then 1 player - with custom notification each player can have separate "notification"

cc @zoriya

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java [hunk 14]
```diff
@@ -174,6 +177,32 @@ public void setSrc(final ReactExoplayerView videoView, @Nullable ReadableMap src
...
+            String description = ReactBridgeUtils.safeGetString(propMetadata, "description");
+            String artist = ReactBridgeUtils.safeGetString(propMetadata, "artist");
+            String imageUriString = ReactBridgeUtils.safeGetString(propMetadata, "imageUri");
+
+            Uri imageUri = null;
+
+            try {
+                imageUri = Uri.parse(imageUriString);
+            } catch (Exception e) {
+                DebugLog.e("ExoPlayer Warning", "Could not parse imageUri in metadata");
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java:194]: @KrzysztofMoch you miss this TAG ! :)

**Review by zoriya — APPROVED** [review]
Not sure how often one would want multiples media notifications at the same times, media controls would get weird too in those cases, but why not. LGTM (even tho I don't know much). Thanks for taking a look at this!

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java (hunk 14 above)
Comment (KrzysztofMoch) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java:194]: We don't have TAG here ...

Comment (freeboub) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java:194]: So create one please 😢

Comment (freeboub) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java:194]: OK, let say it will be for later ... (much more to handle in fact)

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java [hunk 15]
```diff
@@ -174,6 +177,32 @@ public void setSrc(final ReactExoplayerView videoView, @Nullable ReadableMap src
...
+            }
+
+            customMetadata = new MediaMetadata.Builder()
+                    .setTitle(title)
+                    .setSubtitle(subtitle)
+                    .setDescription(description)
+                    .setArtist(artist)
+                    .setArtworkUri(imageUri)
+                    .build();
+        }
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java:204]: I think there should be a else to reset customMetadata, without it when changing content you may keep previous metadata (if no new metadata available)

File: docs/pages/component/props.mdx [hunk 16]
```diff
@@ -694,20 +694,23 @@ source={{ uri: 'https://bitdash-a.akamaihd.net/content/sintel/hls/playlist.m3u8'
 
 #### Overriding the metadata of a source
 
-<PlatformsList types={['tvOS']} />
+<PlatformsList types={['Android, iOS, tvOS']} />
 
-Provide an optional `title`, `subtitle`, `customImageUri` and/or `description` properties for the video.
```
Comment (freeboub) [inline docs/pages/component/props.mdx:699]: be carfull on tvOS, customImageUri is still used ...

Comment (freeboub) [inline docs/pages/component/props.mdx:699]: For for the renaming, can you just ping the original author of this code to inform him of the breaking change please ?

File: ios/Video/RCTVideo.swift [hunk 17]
```diff
@@ -540,26 +563,30 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
...
         }
 
-        if let description = _source?.description {
+        if let description = _source?.customMetadata?.description {
             mapping[.commonIdentifierDescription] = description
         }
 
-        if let customImageUri = _source?.customImageUri,
-           let imageData = RCTVideoUtils.createImageMetadataItem(imageUri: customImageUri) {
+        if let customImageUri = _source?.customMetadata?.imageUri,
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift]:
```suggestion
        if let imageUri = _source?.customMetadata?.imageUri,
```

File: ios/Video/RCTVideo.swift [hunk 18]
```diff
@@ -540,26 +563,30 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
...
+           let imageData = await RCTVideoUtils.createImageMetadataItem(imageUri: customImageUri) {
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift]:
```suggestion
           let imageData = await RCTVideoUtils.createImageMetadataItem(imageUri: imageUri) {
```

**Review by freeboub — CHANGES_REQUESTED** [review]
1 small comment about metadata reset changing source (can be reproduced on android with the sample)
Tested on android, this is great job 👍 
I tried on ios simulator. It was not working at all with the sample (iPhone 15 Pro Max ios 17.4).

+ I will send you small change in the sample for better testing

**Comment by KrzysztofMoch** [comment]
@freeboub just to let you know - all the time I was testing on physic device (iOS)

**Comment by freeboub** [comment]
@KrzysztofMoch the PR: https://github.com/TheWidlarzGroup/react-native-video/pull/5/files

**Comment by freeboub** [comment]
> @freeboub just to let you know - all the time I was testing on physic device (iOS)

can you test on simulator for my curiosity please ? I don't have iPhone anymore :/

**Comment by KrzysztofMoch** [comment]
@Duell10111 ping for you as I made breaking change for your code for custom metadata. I moved propreties for user metadata from `source` to `source.metadata` to make it more readable

**Comment by KrzysztofMoch** [comment]
@freeboub issue with "metadata reset changing source" should be fixed now (now I am disabling notification controls for player with "no media item" (mostly invalid url/source no view etc.)

**Review by freeboub — APPROVED** [review]

**Comment by KrzysztofMoch** [comment]
I have tested iOS simulator and it does not support "NowPlayingInfo" notification so I added to docs info that fot iOS you need to run on real device

**Comment by YangJonghun** [comment]
@KrzysztofMoch
#3723 is awesome! however, prop name seems narrow in scope compared to the actual functionality.
MediaSession is not only about controlling notifications and showing metadata about the video, but it's also about synchronizing processing for external controls (airplay, car, air pod, remote controller etc..).
But it's hard to tell from the prop name if it covers this functionality.

**Comment by KrzysztofMoch** [comment]
@YangJonghun you right, I had not better idea then - we can add this information to docs maybe ?

**Comment by freeboub** [comment]
> @YangJonghun you right, I had not better idea then - we can add this information to docs maybe ?

I agree it is a good idea!

**Comment by zoriya** [comment]
Custom notifications require editing the `AndroidManifest.xml` (as stated in the readme), would it be possible to include those changes on the repository, so expo could automatically apply them? Should I make a new issue to track that?

**Comment by freeboub** [comment]
@zoriya Your request make sens, but I am 100% sure if need, can you please open a ticket or discussion to discuss about it with @KrzysztofMoch ? (or we can continue here ... ).
The issue of integrating it in manifest (move it by default to: android/src/main/AndroidManifest.xml) is that it will make the feature mandatory. 
I don't really care about service declaration (It can be move without side effects). but I worry about new permissions:
```xml
<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
<uses-permission android:name="android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK" />
```

Today, if user doesn't want the feature and don't add it his manifest, will the app still work ?

**Comment by zoriya** [comment]
That's true, it will always request for permissions. I'm not sure if media notifications is a permission that require user consent or if it is automatically granted.

I think this should be opt-out more than opt-in tho, as users, we expect to be able to control media using media buttons.

**Comment by YangJonghun** [comment]
@KrzysztofMoch @freeboub @zoriya 
I think we can make people choose changeable opt-in.

some of the Player's features require permissions, services, and activities.
- service & FOREGROUND_SERVICE(permission) : download, media session, google cast ...
- activity : picture in picture ...

Currently, If someone need these features, they have to write within their AndroidManifest.xml. because Android doesn't support programmatically generate Service or Activity
But, If we write each feature as native Android dependency, people can choose and install with native flag like below
```
    if (ExoplayerDependencies["useReactNativeVideoPIP"]) {
      // For PictureInPIcture support with react-native-video
      implementation "com.react-native-video.picture-in-picture:$rnv_version"
    }
    if (ExoplayerDependencies["useReactNativeVideoMediaSession"]) {
      // For MediaSession support with react-native-video
      implementation "com.react-native-video.media-session:$rnv_version"
    }
```
(As far as I know, `react-native-notifee`'s [app.notifee.core](https://github.com/invertase/notifee/blob/main/android/src/main/AndroidManifest.xml) takes that approach, which allows it to get around some of the limitations of RN library.)

fyi) if we choose this approach, #3385 can be rewrite with Activity.

**Comment by freeboub** [comment]
@YangJonghun Thank you for the clue.
We can also imagine another package react-native-video-media-session which would be responsible of the feature.
BTW, I think the real question we need to answer is should we want to remove mediaSession from the app ?! 

 jetified-media3-session-1.3.1.aar: 506K 
And same question can come for UI:
jetified-media3-ui-1.1.1.aar: 419K

I am not 100% convinced yet of this necessity :/ 
Maybe moving the tags in the manifest can be enough for now ...

**Comment by YangJonghun** [comment]
@freeboub 
I don't care about the size of the package, I want the library to support users to specify only the permissions for the features they actually use (because Play Store has strict permission checks for background behavior).

There are two ways to do this: opt-in and opt-out, and currently we choose opt-out, where users specify the permissions and services they need in their code to use certain features.

My hypothesis is that react-native-video supports opt-in to improve library user's DX. If it's a regular React-Native library, not a native dependency, you can't specify the service, activity, permission, etc. required for the feature separately.

I think opt-in and opt-out are both good!👍

**Comment by vargajacint** [comment]
@YangJonghun You may not but other developers are care about package size

**Comment by freeboub** [comment]
As someone said and remove his comment: "You may not but other developers are care about package size"
That is not the main topic for now :)
Regarding to optin vs optout, I think you choose the good strategy to avoid changing default behavior of the package !
Regarding to the 2 new permissions, I don't think they change anything for store submission even if the feature is not used ? What's why I wonder if we can add it to the AndroidManifest from the package.

**Comment by YangJonghun** [comment]
Recently, Google Play Console has been requiring proof of app content actually using Foreground Service, so if someone not using it and only have the permission added, this could be a reason for rejection.
fyi) https://developer.android.com/about/versions/14/changes/fgs-types-required#google-play-enforcement

**Comment by freeboub** [comment]
@YangJonghun Thank you for the information ! So I think we should keep implementation as is !

**Comment by YangJonghun** [comment]
@KrzysztofMoch 
When the `playInBackground` prop is false, I was expecting the notification control to exit as well, but currently it doesn't behave that way. Is there any plans to change it?

**Comment by KrzysztofMoch** [comment]
On iOS or Android ?

**Comment by YangJonghun** [comment]
@KrzysztofMoch
I tested on Android

**Comment by KrzysztofMoch** [comment]
> @KrzysztofMoch When the `playInBackground` prop is false, I was expecting the notification control to exit as well, but currently it doesn't behave that way. Is there any plans to change it?

I think it makes sense - will try to care of it next week
